File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        saida = {}
        for i, num in enumerate(nums):
            diff = target - num
            if diff in saida:
                return [saida[diff], i]
            saida[num] = i

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const hashMap = {};

    for (let i = 0; i < nums.length; i++) {
        if (target - nums[i] in hashMap) {
            return [i, hashMap[target - nums[i]]];
        }
        hashMap[nums[i]] = i;
    }
};